### DISCUSSION TITLE
How do I run an array of SQL statements?

### DISCUSSION THREAD

**Discussion by skybrian** [porsager/postgres#642] [Q&A]
Hi, I have some code that generates a script for installing some tables. The script is represented as an array of strings where each string is a SQL statement. I'd like to loop over the script and run each statement one at a time.

The tagged literal syntax that Postgres.js uses is cool, but what do I do if I just want to execute a sql statement that's represented as a string?

**Comment by porsager** [comment]
You can either have the script generate queries with ``` sql`` ``` or you can use `sql.unsafe(string)` 😊

**Reply by skybrian** [reply]
Thanks! By "generate queries" do you mean something different from "execute queries?" It might me nice to do some checking of the SQL statements in the script ahead of time, before executing anything.

Also, what's unsafe about sql.unsafe? It looks pretty normal to me, assuming I pass parameters separately. (And I'm just using string constants anyway.)

**Reply by skybrian** [reply]
I tried changing my script type from string[] to postgres.PendingQuery<postgres.MaybeRow[]>[] to see if that would let me use sql as a query builder.

And that fails with: 

```
error: Uncaught (in promise) PostgresError: could not determine data type of placeholder $1
    const error = Errors.postgres(parseError(x))
                         ^
    at ErrorResponse (https://deno.land/x/postgresjs@v3.3.5/src/connection.js:775:26)
    at handle (https://deno.land/x/postgresjs@v3.3.5/src/connection.js:477:6)
    at data (https://deno.land/x/postgresjs@v3.3.5/src/connection.js:318:9)
    at https://deno.land/x/postgresjs@v3.3.5/polyfills.js:150:32
    at Array.forEach (<anonymous>)
    at call (https://deno.land/x/postgresjs@v3.3.5/polyfills.js:150:18)
    at success (https://deno.land/x/postgresjs@v3.3.5/polyfills.js:110:11)
    at eventLoopTick (ext:core/01_core.js:183:11)

```

I don't get it so I reverted to string[].
